wordpress-mobile/WordPress-FluxC-Android
# Use a single call for XML-RPC media uploads (when able)

Addresses #423. The response of `wp.uploadFile` is checked, and if it contains the required fields (should always be present on WordPress 4.4 and later) it updates the `MediaModel` using those and doesn't call `wp.getMediaItem`.

The client falls back to the old behavior (`wp.uploadFile` + `wp.getMediaItem`) if any fields are missing from the upload response.

Also added a new test site for a WordPress 4.3 site and some tests so we can catch any future regressions.

For reference, this is the core change that modified `wp.uploadFile` to respond similarly to `wp.getMediaItem`: https://core.trac.wordpress.org/changeset/34579

To test:
* Run the old and new upload tests with network sniffing, and verify that modern enough WordPress sites complete the upload with a single call.
* Always return `false` from [`isDeprecatedUploadResponse()`](https://github.com/wordpress-mobile/WordPress-FluxC-Android/compare/feature/423-single-call-xmlrpc-uploads?expand=1#diff-a59c679264e9e17f7dca0a78844f29d2R519) and watch the WordPress 4.3 tests fail
